src/cli/api/remote_api.py

```diff
@@ -1,69 +1,150 @@
 from pathlib import Path
-from typing import List, Optional
+import time
+from typing import Any, List, Optional, cast
 
 from cli.managers.config_manager import ConfigManager
-from cli.types import AppConfigType, ApplicationType, ResultType
+from cli.managers.auth_manager import AuthManager
+from cli.type_aliases import AppConfigType, ApplicationType, ExperimentType, ResultType
+from cli.utils import write_json_file
 
 
 class RemoteApi:
     def __init__(self, config_manager: ConfigManager) -> None:
-        self.__config_manager = config_manager
+        self.__config_manager: ConfigManager = config_manager
+        config_dir = self.__config_manager.get_config_dir()
+        self.auth_manager:AuthManager = AuthManager(str(config_dir.resolve()),
+                                                    self.__login_user, self.__login_anonymous)
 
     def login(self, username: str, password: str, host: str) -> None:
-        pass
+        self.auth_manager.login(username=username, password=password, host=host)
 
     def __login_user(self, username: str, password: str, host: str) -> str:
         pass
 
     def __login_anonymous(self) -> str:
         pass
 
-    def logout(self) -> None:
-        pass
+    def logout(self, host: str) -> None:
+        if host is None:
+            host = self.auth_manager.get_active_host()
+
+        self.auth_manager.delete_token(host)
 
     def list_applications(self) -> List[ApplicationType]:
-        pass
+        token = self.auth_manager.load_token()
+        self._action('listApplications', {'token': token})
+        return []
 
     def delete_application(self, application: str) -> None:
-        pass
+        app = self.__config_manager.get_application(application)
+        token = self.auth_manager.load_token()
+        self._action('delete', {'token': token, 'id': app})
 
     def upload_application(self, application: str) -> None:
-        pass
+        token = self.auth_manager.load_token()
```

Comment:
I think we should solve this differently because the token handling should be done at 1 place instead of all the remote methods separately. Since we don't do remote I wonder why we added this code already. the `pass` as it was was enough wasn't it. Same for the rest of this file